### ISSUE TITLE
Prefer final function parameters

### ISSUE THREAD

**Issue by Scriptim** [dart-lang/linter#1914]
**Describe the rule you'd like to see implemented**
Some people, including myself, prefer `final` function parameters. A linter rule `prefer_final_function_parameters` should detect when a parameter is not `final`.

I don't know if such a rule makes sense and if there are cases where you should not make function parameters `final`.

**Example**
```dart
	public void foo(String bar) {} 		  // invalid
	public void foo(final String bar) {}  // valid
```

**Comment by NatoBoram** [comment]
Function parameters are technically local variables since they're passed by value, aren't they? I feel like `prefer_final_locals` should handle these as well.
